paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/FlinkJobRecoveryITCase.java

```diff
@@ -0,0 +1,320 @@
+            batchSql(sql);
+        }
+
+        Configuration config =
+                sEnv.getConfig()
+                        .getConfiguration()
+                        .set(StateRecoveryOptions.SAVEPOINT_PATH, checkpointPath);
+        for (Map.Entry<String, String> entry : recoverOptions.entrySet()) {
+            config.setString(entry.getKey(), entry.getValue());
+        }
+
+        jobClient = sEnv.executeSql(afterRecoverSql).getJobClient().get();
+        triggerCheckpointAndWaitForWrites(jobClient, afterRecoverExpectedRows.size());
+        jobClient.cancel().get();
+
+        rows = batchSql(afterRecoverCheckSql);
+        assertThat(rows.size()).isEqualTo(afterRecoverExpectedRows.size());
+        assertThat(rows).containsExactlyInAnyOrder(afterRecoverExpectedRows.toArray(new Row[0]));
+    }
+
+    private void createTargetTable(String tableName, BucketMode bucketMode) {
+        switch (bucketMode) {
+            case HASH_FIXED:
+                batchSql(
+                        String.format(
+                                "CREATE TABLE IF NOT EXISTS `%s` (k INT, f1 STRING, pt STRING, PRIMARY KEY(k, pt) NOT ENFORCED) WITH ('bucket'='2', 'commit.force-create-snapshot'='true')",
+                                tableName));
+                return;
+            case HASH_DYNAMIC:
+                batchSql(
+                        String.format(
+                                "CREATE TABLE IF NOT EXISTS `%s` (k INT, f1 STRING, pt STRING, PRIMARY KEY(k, pt) NOT ENFORCED) WITH ('bucket'='-1', 'commit.force-create-snapshot'='true')",
+                                tableName));
+                return;
+            case CROSS_PARTITION:
+                batchSql(
+                        String.format(
+                                "CREATE TABLE IF NOT EXISTS `%s` (k INT, f1 STRING, pt STRING, PRIMARY KEY(k) NOT ENFORCED) WITH ('bucket'='-1', 'commit.force-create-snapshot'='true')",
+                                tableName));
+                return;
+            case BUCKET_UNAWARE:
+                batchSql(
+                        String.format(
+                                "CREATE TABLE IF NOT EXISTS `%s` (k INT, f1 STRING, pt STRING) WITH ('bucket'='-1', 'commit.force-create-snapshot'='true')",
+                                tableName));
+                return;
+            default:
+                throw new IllegalArgumentException("Unsupported bucket mode: " + bucketMode);
+        }
+    }
+
+    private Snapshot waitForNewSnapshot(String tableName, long initialSnapshot)
+            throws InterruptedException {
+        Snapshot snapshot = findLatestSnapshot(tableName);
+        while (snapshot == null || snapshot.id() == initialSnapshot) {
+            Thread.sleep(2000L);
+            snapshot = findLatestSnapshot(tableName);
+        }
+        return snapshot;
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T> T reflectGetField(Object instance, String fieldName)
```

Comment:
Actual value of parameter 'fieldName' is always 'miniCluster' 

Private static final String xxx = "miniCluster";